src/core/services/administrator_service.py

```diff
@@ -0,0 +1,95 @@
+    InvalidAuthenticationDataException,
+    UnauthorizedException,
+)
+from src.core.settings import settings
+
+PASSWORD_CONTEXT = CryptContext(schemes=["bcrypt"], deprecated="auto")
+OAUTH2_SCHEME = OAuth2PasswordBearer(tokenUrl="/administrators/login", scheme_name="JWT")
+
+ACCESS_TOKEN_EXPIRE_MINUTES = 30
+REFRESH_TOKEN_EXPIRE_MINUTES = 60 * 24 * 10
+ALGORITHM = "HS256"
+
+
+class AdministratorService:
+    def __init__(self, administrator_repository: AdministratorRepository = Depends()):
+        self.__administrator_repository = administrator_repository
+
+    def __get_hashed_password(self, password: str) -> str:
+        """Получить хэш пароля."""
+        return PASSWORD_CONTEXT.hash(password)
+
+    def __verify_hashed_password(self, plain_password: str, hashed_password: str) -> bool:
+        """Сравнить открытый пароль с хэшем."""
+        return PASSWORD_CONTEXT.verify(plain_password, hashed_password)
+
+    def __create_jwt_token(self, subject: str, expires_delta: int) -> str:
+        """Создать jwt-токен.
+
+        Аргументы:
+            subject (str) - субьект токена
+            expires_delta (int) - время жизни токена
+        """
+        expire = dt.datetime.utcnow() + dt.timedelta(minutes=expires_delta)
+        to_encode = {"sub": subject, "exp": expire}
+        return jwt.encode(to_encode, settings.SECRET_KEY, ALGORITHM)
+
+    async def get_administrator_by_email(self, email: str) -> Administrator:
+        """Получить администратора по его email."""
+        return await self.__administrator_repository.get_by_email_or_none(email)
+
+    async def __authenticate_administrator(self, auth_data: AdministratorAuthenticateRequest) -> Administrator:
+        """Аутентификация администратора по email и паролю."""
+        administrator = await self.get_administrator_by_email(auth_data.email)
+        if not administrator:
+            raise InvalidAuthenticationDataException()
+        if administrator.status == Administrator.Status.BLOCKED:
+            raise AdministratorBlockedException()
+        password = auth_data.password.get_secret_value()
+        if not self.__verify_hashed_password(password, administrator.hashed_password):
+            raise InvalidAuthenticationDataException()
+        return administrator
+
+    async def get_access_and_refresh_tokens(self, auth_data: AdministratorAuthenticateRequest) -> TokenResponse:
+        """Получить access и refresh токены."""
+        administrator = await self.__authenticate_administrator(auth_data)
+        administrator.last_logined_at = dt.datetime.now()
+        await self.__administrator_repository.update(administrator.id, administrator)
+
+        subject = administrator.email
+        tokens = {
+            "access_token": self.__create_jwt_token(subject, ACCESS_TOKEN_EXPIRE_MINUTES),
+            "refresh_token": self.__create_jwt_token(subject, REFRESH_TOKEN_EXPIRE_MINUTES),
+        }
+        return TokenResponse(**tokens)

```

Comment:
Можно и без промежуточного словаря.
```suggestion
        return TokenResponse(
            access_token = self.__create_jwt_token(subject, ACCESS_TOKEN_EXPIRE_MINUTES),
            refresh_token = self.__create_jwt_token(subject, REFRESH_TOKEN_EXPIRE_MINUTES),
        )
```